### DISCUSSION TITLE
Proposal:Batch create privileges for menuitems.

### DISCUSSION THREAD

**Discussion by ZhaoqiangCn** [TrudAX/TRUDUtilsD365#28] [Ideas]
Sometimes we forget to create privileges for menu items during development. It would be great if we could create them in batches.

**Comment by TrudAX** [comment]
Thanks for the proposal @ZhaoqiangCn . 
How we currently do this:
Menu items without privilege will cause BP error. So how the flow is currently implemented on our project: Every developer is required to check BP after the modification is finished(we disable all non relevant checks). If you have a Menu items without privilege it will be displayed in the output. So you click on it, and there is util to create a privilege(you still need to check/adjust label, select the access level)
If you have an existing project and you want to create all missing privileges, just run a compile, and start fixing all BP errors, I don't think any util helps in this. This step is actually required before GoLive, you need to submit a CAR report.

Let me know more about your case and why it can't be resolved using existing tools

**Comment by ZhaoqiangCn** [comment]
> Thanks for the proposal @ZhaoqiangCn . How we currently do this: Menu items without privilege will cause BP error. So how the flow is currently implemented on our project: Every developer is required to check BP after the modification is finished(we disable all non relevant checks). If you have a Menu items without privilege it will be displayed in the output. So you click on it, and there is util to create a privilege(you still need to check/adjust label, select the access level) If you have an existing project and you want to create all missing privileges, just run a compile, and start fixing all BP errors, I don't think any util helps in this. This step is actually required before GoLive, you need to submit a CAR report.
> 
> Let me know more about your case and why it can't be resolved using existing tools

Since we're working on an upgrade project and many buttons were migrated, they currently lack necessary privileges.

**Comment by TrudAX** [comment]
Ok, got it. However, if you upgrade from AX2012, the LCS upgrade tool should create old privileges. If you talk about buttons, one form of privilege probably contains multiple menuitems, so this still require some manual work, and the best way to approach it - run compile with BP check and process one by one.
If you want just to create privileges in a bulk, the best way is probably make several examples and ask ChatGPT to write a script to generate XML for all files in folder, based on provided examples. it is a one-time task, if you do it this way, it will be good that you share it here
